### PR TITLE
fix uniqueness by id check, handle missing data

### PR DESCRIPTION
**Pull request by brad**
@orcas/orcas-developers Please review. The updates:

- Handle possibly missing 'name' and 'avatar' from the profile API
- Handle users with no devices
- Handle goals with no ids
- Dedupe goals and sessions by id instead of date
- pep8 updates
- fix missing coverage data. Files were being imported before coverage was enabled, so class/method definitions were not showing up as covered

### CODE REVIEW

File: misfitapp/models.py [hunk 1]
```diff
@@ -150,11 +154,11 @@ def import_from_misfit(cls, misfit, uid, object_id=None):
             'email': profile.email,
             'birthday': profile.birthday,
             'gender': profile.gender,
-            'name': profile.name
+            # These two attributes aren't always included in the API results
+            # despite the fact that the docs say they are not optional
```
Comment (grokcode) [inline misfitapp/models.py]: @brad Thanks for commenting this discrepancy.

File: misfitapp/models.py [hunk 2]
```diff
@@ -227,18 +237,21 @@ def import_all_from_misfit(cls, misfit, uid,
...
             goals = misfit.goal(start_date=start, end_date=end)
             for goal in goals:
+                if not hasattr(goal, 'id'):
+                    # For some reason, goals occasionally have no id, ignore
+                    continue
                 if goal.id not in exists:
                     model_data = cls.data_dict(goal)
                     obj_list.append(cls(user_id=uid, **model_data))
-        cls.objects.bulk_create(dedupe_by_field(obj_list, 'date'))
+        cls.objects.bulk_create(dedupe_by_field(obj_list, 'id'))
```
Comment (percyperez) [inline misfitapp/models.py]: @brad Good use of `bulk_create`

File: test_settings.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    },
+    'loggers': {
+        'misfitapp.tasks': {'handlers': ['console'], 'level': 'DEBUG'},
+    },
+}
+
+MIDDLEWARE_CLASSES = (
+    'django.contrib.sessions.middleware.SessionMiddleware',
+    'django.contrib.auth.middleware.AuthenticationMiddleware',
+)
```
Comment (grokcode) [inline test_settings.py]: @brad Thanks for this change, it is nice to have this in it's own file. 👍

**Comment by grokcode** [comment]
Looks great! :shipit:

**Comment by percyperez** [comment]
LGMT2

**Pull request merged**